src/server/utilities.ts

```diff
@@ -17,22 +17,11 @@ namespace ts.server {
         loggingEnabled(): boolean;
         perftrc(s: string): void;
         info(s: string): void;
-        startGroup(): void;
-        endGroup(): void;
-        msg(s: string, type?: Msg.Types): void;
+        err(s: string): void;
+        group(logGroupEntries: (log: (msg: string) => void) => void): void;
         getLogFileName(): string;
     }
 
-    export namespace Msg {

```

Comment:
There should only be one reference to each kind of message -- on each in `perftrc()`, `info()`, and `err()`.